classes/Infrastructure/Event/ExceptionListener.php

```diff
@@ -46,40 +45,25 @@ public function onKernelException(GetResponseForExceptionEvent $event)
         $exception = $event->getException();
         $request   = $event->getRequest();
 
-        $event->setResponse($this->renderResponse($request, $exception));
-    }
-
-    private function renderResponse(Request $request, \Throwable $exception): Response
-    {
-        if (\in_array('application/json', $request->getAcceptableContentTypes())) {
-            $headers = [];
-
-            if ($exception instanceof HttpExceptionInterface) {
-                $code    = $exception->getStatusCode();
-                $headers = $exception->getHeaders();
-            }
-
-            return new JsonResponse([
-                'error' => $exception->getMessage(),
-            ], $code, $headers);
+        if (!\in_array('application/json', $request->getAcceptableContentTypes())) {
+            return;
         }
 
-        $template = 'error/500.twig';
-
-        $templates = [
-            Response::HTTP_UNAUTHORIZED => 'error/401.twig',
-            Response::HTTP_FORBIDDEN    => 'error/403.twig',
-            Response::HTTP_NOT_FOUND    => 'error/404.twig',
-        ];
+        $event->setResponse($this->renderResponse($exception));
+    }
 
-        $code = $exception instanceof HttpExceptionInterface ? $exception->getStatusCode() : 500;
+    private function renderResponse(\Throwable $exception): Response
+    {
+        $code    = 500;
+        $headers = [];
 
-        if (\array_key_exists($code, $templates)) {
-            $template = $templates[$code];
+        if ($exception instanceof HttpExceptionInterface) {
+            $code    = $exception->getStatusCode();
+            $headers = $exception->getHeaders();
         }
 
-        $message = $this->twig->render($template);
-
-        return new Response($message, $code);
+        return new JsonResponse([
+            'error' => $exception->getMessage(),
+        ], $code, $headers);
```

Comment:
Not a big deal, but at first glance it's not quite apparent that we actually pass three arguments to the constructor of `JsonResponse`, maybe we can adjust the wrapping a bit to make that more obvious?

```diff
diff --git a/classes/Infrastructure/Event/ExceptionListener.php b/classes/Infrastructure/Event/ExceptionListener.php
index a1581611..b51c1b2e 100644
--- a/classes/Infrastructure/Event/ExceptionListener.php
+++ b/classes/Infrastructure/Event/ExceptionListener.php
@@ -62,8 +62,12 @@ class ExceptionListener implements EventSubscriberInterface
             $headers = $exception->getHeaders();
         }

-        return new JsonResponse([
-            'error' => $exception->getMessage(),
-        ], $code, $headers);
+        return new JsonResponse(
+            [
+                'error' => $exception->getMessage(),
+            ],
+            $code,
+            $headers
+        );
     }
 }
```